### PR TITLE
Create a document on how to create a Go nested module

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

- Document how to create a new Go module.

### Motivation

- Have a public guideline on the necessary steps.

### CODE REVIEW

**Review by derekwbrown — APPROVED** [review]

File: docs/dev/modules.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+1. Create a `tools.go` file at the module root folder. You can use the one on the main folder as a starting point. This will define the tools used for CI.
+1. Create a `.golangci.yml` file at the module root folder. You can use the one on the main folder as a starting point. This defines how the `golangci-lint` CI check works.
+1. Create the `go.mod` and `go.sum` files at the module root folder. You can use `go mod init && go mod tidy` within the folder of the module as a starting point. Make sure the go version line matches the version on the main `go.mod` file.
+1. On each module that depends on the current one, add a `require` directive with the module path with version `v0.0.0` 
+1. On each module that depends on the current one, add a `replace` directive to replace the module with the repository where it lives.
+1. Update the `DEFAULT_MODULES` dictionary on the `tasks/modules.py` file. You need to:
+    - create a new module, specifying the path, targets, dependencies and a condition to run tests (if any) and
+    - update the dependencies of other modules if they depend on this one.
```
Comment (kayayarai) [inline docs/dev/modules.md:19]: This feels like it could use an example or end result example to show the reader what it should look like.

**Review by kayayarai — COMMENTED** [review]
Some suggestions for grammar and clarity.

File: docs/dev/modules.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+At present the Go modules offer no public stability guarantees and are intended for internal consumption by Datadog only.
+
+## Creating a new Go nested module
+
+To create a new Go nested module, you will first need to determine the packages that you want to expose and their dependencies. You might need to refactor the code to have an exportable package, since the `replace` directives that we use might be incompatible with your project.
+After the possible refactor and once you have listed all the packages that you want to expose and their dependencies, you will need to follow these steps for each module you want to create.
+
+1. Create a `tools.go` file at the module root folder. You can use the one on the main folder as a starting point. This will define the tools used for CI.
+1. Create a `.golangci.yml` file at the module root folder. You can use the one on the main folder as a starting point. This defines how the `golangci-lint` CI check works.
```
[the remark this answers is not in the document]
Comment (mx-psi) [reply docs/dev/modules.md:13]: Argh, thanks, choosing the right preposition is one of the hardest things of writing/speaking in English for me :)

File: docs/dev/modules.md (hunk 1 above)
Comment (mx-psi) [reply docs/dev/modules.md:32]: Done!

File: docs/dev/modules.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+1. Create the `go.mod` and `go.sum` files at the module root folder. You can use `go mod init && go mod tidy` within the folder of the module as a starting point. Make sure the go version line matches the version on the main `go.mod` file.
+1. On each module that depends on the current one, add a `require` directive with the module path with version `v0.0.0` 
+1. On each module that depends on the current one, add a `replace` directive to replace the module with the repository where it lives.
```
[the remark this answers is not in the document]
Comment (mx-psi) [reply docs/dev/modules.md]: I added a code example for each of them

**Comment by mx-psi** [comment]
Thanks for the useful suggestions @kayayarai! I think I adressed everything, could you have another look?

**Review by kayayarai — APPROVED** [review]
Looks great!

**Pull request merged**